### PR TITLE
add snippets for automatic batching

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - Python snippets (based on ov_auto_branching.cpp) for automatic branching page

### Tickets:
 - 80644

### CODE REVIEW

File: docs/snippets/ov_auto_batching.py [hunk 1]
```diff
@@ -1,41 +1,33 @@
...
+model = core.read_model(model="sample.xml")
 
-//! [compile_model]
-{
-    auto compiled_model = core.compile_model(model, "GPU", ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT));
-}
-//! [compile_model]
+# [compile_model]
+config = {"PERFORMANCE_HINT": "THROUGHPUT"}
+compiled_model = core.compile_model(model, "GPU", config)
```
Comment (ilya-lavrenov) [inline docs/snippets/ov_auto_batching.py:8]: why config is not passed as `core.compile_model(model, "GPU", config={"PERFORMANCE_HINT": "THROUGHPUT"})`?

File: docs/snippets/ov_auto_batching.py [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
-    // leaving intact other configurations options that the device selects for the 'throughput' hint 
-    auto compiled_model = core.compile_model(model, "GPU", {ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT),
-                                                            ov::hint::allow_auto_batching(false)});
-}
-//! [compile_model_no_auto_batching]
+# [compile_model_no_auto_batching]
+# disabling the automatic batching
+# leaving intact other configurations options that the device selects for the 'throughput' hint 
+config = {"PERFORMANCE_HINT": "THROUGHPUT",
+          "ALLOW_AUTO_BATCHING": False}
```
Comment (ilya-lavrenov) [inline docs/snippets/ov_auto_batching.py:15]: does `false` work? I expect "NO"

**Comment by ilya-lavrenov** [comment]
BTW, don't forget about releases/2022/1 branch

File: docs/snippets/ov_auto_batching.py (hunk 2 above)
Comment (bszmelcz) [reply docs/snippets/ov_auto_batching.py:15]: yup, should be "NO"

File: docs/snippets/ov_auto_batching.py (hunk 1 above)
Comment (bszmelcz) [reply docs/snippets/ov_auto_batching.py:8]: Just my personal preference but I am gonna change it for your approach

Comment (ilya-lavrenov) [reply docs/snippets/ov_auto_batching.py:8]: ok, we can use different approaches in different examples to demonstrate all the variants.

**Comment by AlexeyLebedev1** [comment]
Please also add a left bracket here https://github.com/openvinotoolkit/openvino/commit/180f15e84c1bb30f4c6222034dc80453882dcb0f#diff-837b3c7d4bf06daa50c3a81617b5a5c0dd18b49dca8fb493eaff927b448f1458R79

File: docs/snippets/ov_auto_batching.py [hunk 3]
```diff
@@ -1,41 +1,33 @@
...
-{
-    // when the batch size is automatically selected by the implementation
-    // it is important to query/create and run the sufficient #requests
-    auto compiled_model = core.compile_model(model, "GPU", ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT));
-    auto num_requests = compiled_model.get_property(ov::optimal_number_of_infer_requests);
-}
-//! [query_optimal_num_requests]
+# [query_optimal_num_requests]
+# when the batch size is automatically selected by the implementation
+# it is important to query/create and run the sufficient #requests
```
Comment (AlexeyLebedev1) [inline docs/snippets/ov_auto_batching.py:21]:
```suggestion
# it is important to query/create and run the sufficient requests
```

**Review by AlexeyLebedev1 — CHANGES_REQUESTED** [review]

**Comment by ilya-lavrenov** [comment]
is it still draft? can be merged?

**Review by AlexeyLebedev1 — APPROVED** [review]

**Pull request merged**